### PR TITLE
SDK-231: Staged execution

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
[SDK-231](https://spryker.atlassian.net/browse/SDK-231)

### CODE REVIEW

File: src/Entity/ContextInterface.php [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+     * @return array<\SprykerSdk\SdkContracts\Entity\MessageInterface>
+     */
+    public function getMessages(): array;
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\MessageInterface $message
+     *
+     * @return void
+     */
+    public function addMessage(MessageInterface $message);
```
Comment (gechetspr) [inline src/Entity/ContextInterface.php]: mb we can add return type here?

File: src/Entity/ContextInterface.php [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+     * @return array<\SprykerSdk\SdkContracts\Entity\TaskInterface>
+     */
+    public function getSubTasks(): array;
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\TaskInterface $task
+     *
+     * @return void
+     */
+    public function addSubTask(TaskInterface $task);
```
Comment (gechetspr) [inline src/Entity/ContextInterface.php]: same for all void return types

**Review by gechetspr — CHANGES_REQUESTED** [review]

File: src/Entity/ContextInterface.php (hunk 1 above)
Comment (Kyrylo-Meretskyi) [reply src/Entity/ContextInterface.php]: Fixed

File: src/Entity/ContextInterface.php (hunk 2 above)
Comment (Kyrylo-Meretskyi) [reply src/Entity/ContextInterface.php]: Fixed

**Review by vol4onok — APPROVED** [review]

**Review by newLoki — APPROVED** [review]

File: src/Entity/ContextInterface.php (hunk 1 above)
Comment (vol4onok) [reply src/Entity/ContextInterface.php]: update message structure array<string, MessageInterface> key will be id of command, also create errorCode properties as Messages. Errors should have array<string, int> structure

Comment (vol4onok) [reply src/Entity/ContextInterface.php]: update message structure array<string, MessageInterface> key will be id of command, also create errorCode properties as Messages. Errors should have array<string, int> structure

**Review by gechetspr — APPROVED** [review]

File: src/Entity/ContextInterface.php [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+     * @return int
+     */
+    public function getExitCode(): int;
+
+    /**
+     * @param int $exitCode
+     *
+     * @return void
+     */
+    public function setExitCode(int $exitCode): void;
```
Comment (vol4onok) [inline src/Entity/ContextInterface.php:132]: it also should be multiple like message and addExistCode() method

**Pull request merged**